pkg/fleet-manager/backup_controller_helper.go

```diff
@@ -0,0 +1,490 @@
+}
+
+// isFleetClusterSubset is the helper function to check if one set of clusters is a subset of another
+func isFleetClusterSubset(baseClusters, subsetClusters map[ClusterKey]*fleetCluster) bool {
+	for key := range subsetClusters {
+		if _, exists := baseClusters[key]; !exists {
+			return false
+		}
+	}
+	return true
+}
+
+func (b *BackupManager) fetchMigrateSourceClusters(ctx context.Context, namespace string, destination backupapi.Destination) (sourceClusterName, sourceClusterNameKind string, sourceCluster *fleetCluster, err error) {
+	fleetClusters, err := b.fetchDestinationClusters(ctx, namespace, destination)
+	if err != nil {
+		return "", "", nil, err
+	}
+
+	// SourceCluster must contain one Clusters, it is ensured by admission webhook
+	for key, value := range fleetClusters {
+		sourceClusterName = key.Name
+		sourceClusterNameKind = key.Kind
+		sourceCluster = value
+	}
+
+	return sourceClusterName, sourceClusterNameKind, sourceCluster, err
+}
+
+// buildVeleroBackupInstance constructs a Velero Backup instance configured to perform a backup operation on the specified cluster.
+func buildVeleroBackupInstance(backupSpec *backupapi.BackupSpec, labels map[string]string, veleroBackupName string) *velerov1.Backup {
+	veleroBackup := &velerov1.Backup{
+		ObjectMeta: generateVeleroResourceObjectMeta(veleroBackupName, labels),
+		Spec:       buildVeleroBackupSpec(backupSpec.Policy),
+	}
+	return veleroBackup
+}
+
+// buildVeleroBackupInstanceUsingMigrate constructs a Velero Backup instance configured to perform a backup operation on the specified cluster using migrate config.
+func buildVeleroBackupInstanceUsingMigrate(migrateSpec *backupapi.MigrateSpec, labels map[string]string, veleroBackupName string) *velerov1.Backup {
+	// when using velero backup, only care about migrateSpec.Policy.ResourceFilter
+	backupParam := &backupapi.BackupSpec{}
+	if migrateSpec.Policy != nil {
+		backupParam.Policy = &backupapi.BackupPolicy{}
+		backupParam.Policy.ResourceFilter = migrateSpec.Policy.ResourceFilter
+	}
+	return buildVeleroBackupInstance(backupParam, labels, veleroBackupName)
+}
+
+// buildVeleroScheduleInstance constructs a Velero Schedule instance configured to schedule backup operations on the specified cluster.
+func buildVeleroScheduleInstance(backupSpec *backupapi.BackupSpec, labels map[string]string, veleroBackupName string) *velerov1.Schedule {
+	veleroSchedule := &velerov1.Schedule{
+		ObjectMeta: generateVeleroResourceObjectMeta(veleroBackupName, labels),
+		Spec: velerov1.ScheduleSpec{
+			Template: buildVeleroBackupSpec(backupSpec.Policy),
+			Schedule: backupSpec.Schedule,
+		},
+	}
+	return veleroSchedule
+}
+
+// buildVeleroScheduleInstance constructs a Velero Restore instance configured to Restore operations on the specified cluster.
+func buildVeleroRestoreInstance(restoreSpec *backupapi.RestoreSpec, labels map[string]string, veleroBackupName, veleroRestoreName string) *velerov1.Restore {
```

Comment:
Can we remove all `instance` suffixes, `VeleroRestore` is very clear to me

Reply:
ok